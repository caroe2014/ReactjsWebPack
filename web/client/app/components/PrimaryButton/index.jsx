// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { Button } from 'rebass';
import styled from 'styled-components';

const StyledButton = styled(Button)`
  background-color: ${props => props.theme.colors.buttonControlColor};
  border-radius: 0;
  color: ${props => props.theme.colors.buttonTextColor};
  cursor: pointer;
  font-size: ${props => props.theme.fontSize.md};
  font-weight: 500;
  padding: 20px;
  text-transform: uppercase;
  width: 100%;
  box-shadow: none;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

export default function PrimaryButton (props) {
  const { classContext, tabIndex, ariaLabel, ...rest } = props;
  return (
    <StyledButton className={classContext} {...rest}
      tabIndex={tabIndex}
      aria-label={ariaLabel}>
      {props.children}
    </StyledButton>
  );
}
