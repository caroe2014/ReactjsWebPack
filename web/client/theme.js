// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
