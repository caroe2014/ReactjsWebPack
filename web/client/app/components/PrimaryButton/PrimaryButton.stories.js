// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Provider } from 'rebass';
import theme from 'web/client/theme.js';
import PrimaryButton from '.';

storiesOf('Button', module)
  .add('PrimaryButton', () =>
    <Provider theme={theme} style={{ height: '250px' }}>
      <PrimaryButton onClick={action('clicked')} children='Button'/>
    </Provider>
  );
