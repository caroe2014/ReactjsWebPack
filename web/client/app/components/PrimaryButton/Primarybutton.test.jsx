import React from 'react';
import {
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import theme from 'web/client/theme';

import PrimaryButton from '.';

configure({
  adapter: new Adapter()
});

describe('PrimaryButton', () => {

  it('does a snapshot test', () => {
    const wrapper = mount(<PrimaryButton classContext='test'  theme={theme} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(mount(<PrimaryButton classContext='test'  theme={theme} />).exists()).toBe(true);
  });
});
