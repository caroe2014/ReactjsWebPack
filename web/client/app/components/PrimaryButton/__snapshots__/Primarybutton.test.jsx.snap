// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton does a snapshot test 1`] = `
.c1 {
  font-family: inherit;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: 14px;
  font-weight: bold;
  line-height: 1.1428571428571428;
  margin: 0px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: blue;
  border-radius: 2px;
  border: 0;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:focus {
  outline: none;
  box-shadow: 0 0 0 2px blue;
}

.c0 {
  border-radius: 0;
  cursor: pointer;
  font-size: 1.3rem;
  font-weight: 500;
  padding: 20px;
  text-transform: uppercase;
  width: 100%;
  box-shadow: none;
}

.c0:focus {
  outline: none;
  box-shadow: none;
}

<PrimaryButton
  classContext="test"
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <Styled(Button)
    className="test"
    theme={
      Object {
        "colors": Object {
          "addOnTab": "#347CFF",
          "addToCart": "#347CFF",
          "addonBg": "#EEEEEE",
          "backgroundGrey": "#E6E6E6",
          "black": "#000000",
          "close": "red",
          "contextBackground": "#f0f0f3",
          "darkGrey": "#464646",
          "disableBackground": "#E4E4E4",
          "disableButton": "#D1D1D1",
          "error": "#cc0000",
          "footer": "#48a942",
          "glutenFree": "#8d7356",
          "healthy": "#359919",
          "hint": "#a6a6a6",
          "inputBottomBorder": "lightgray",
          "itemBorder": "lightgrey",
          "itemTile": "#3C474C0",
          "light": "white",
          "loader": "#ababab",
          "loginBackground": "#303030",
          "loginBoxColor": "#434149",
          "loginField": "#AAA7AC",
          "loginFormBackground": "#484848",
          "navigationMenu": "#48a942",
          "navigationMenuHover": "cornflowerblue",
          "new": "#ff6000",
          "open": "#48a942",
          "paymentSuccessTitle": "#505050",
          "placeHolder": "#909090",
          "primary": "#347CFF",
          "primaryButton": "#347CFF",
          "primaryText": "#383838",
          "secondaryText": "#a6a6a6",
          "spicy": "#ed1c24",
          "text": "black",
          "textGrey": "#505050",
          "tile": "#505050",
          "validationError": "#cc0000",
          "vegan": "#0d4a22",
          "vegetarian": "#0d4a22",
        },
        "fontFamily": "Roboto",
        "fontSize": Object {
          "lg": "1.8rem",
          "md": "1.3rem",
          "nm": "1rem",
          "sm": "0.75rem",
          "xl": "2.4rem",
        },
        "fontSizeFromProps": [Function],
        "mediaBreakpoints": Object {
          "desktop": [Function],
          "mobile": [Function],
          "tablet": [Function],
        },
        "nav": Object {
          "height": "70px",
        },
        "receiptFontFamily": "Roboto",
        "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
        "spacing": Object {
          "lg": "4em",
          "md": "2em",
          "nm": "1em",
          "none": "0em",
          "sm": "0.5em",
          "xl": "8em",
        },
      }
    }
  >
    <Button
      bg="blue"
      border={0}
      borderRadius={2}
      className="test c0"
      color="white"
      fontSize={1}
      fontWeight="bold"
      is="button"
      lineHeight={1.1428571428571428}
      m={0}
      px={3}
      py={2}
      theme={
        Object {
          "colors": Object {
            "addOnTab": "#347CFF",
            "addToCart": "#347CFF",
            "addonBg": "#EEEEEE",
            "backgroundGrey": "#E6E6E6",
            "black": "#000000",
            "close": "red",
            "contextBackground": "#f0f0f3",
            "darkGrey": "#464646",
            "disableBackground": "#E4E4E4",
            "disableButton": "#D1D1D1",
            "error": "#cc0000",
            "footer": "#48a942",
            "glutenFree": "#8d7356",
            "healthy": "#359919",
            "hint": "#a6a6a6",
            "inputBottomBorder": "lightgray",
            "itemBorder": "lightgrey",
            "itemTile": "#3C474C0",
            "light": "white",
            "loader": "#ababab",
            "loginBackground": "#303030",
            "loginBoxColor": "#434149",
            "loginField": "#AAA7AC",
            "loginFormBackground": "#484848",
            "navigationMenu": "#48a942",
            "navigationMenuHover": "cornflowerblue",
            "new": "#ff6000",
            "open": "#48a942",
            "paymentSuccessTitle": "#505050",
            "placeHolder": "#909090",
            "primary": "#347CFF",
            "primaryButton": "#347CFF",
            "primaryText": "#383838",
            "secondaryText": "#a6a6a6",
            "spicy": "#ed1c24",
            "text": "black",
            "textGrey": "#505050",
            "tile": "#505050",
            "validationError": "#cc0000",
            "vegan": "#0d4a22",
            "vegetarian": "#0d4a22",
          },
          "fontFamily": "Roboto",
          "fontSize": Object {
            "lg": "1.8rem",
            "md": "1.3rem",
            "nm": "1rem",
            "sm": "0.75rem",
            "xl": "2.4rem",
          },
          "fontSizeFromProps": [Function],
          "mediaBreakpoints": Object {
            "desktop": [Function],
            "mobile": [Function],
            "tablet": [Function],
          },
          "nav": Object {
            "height": "70px",
          },
          "receiptFontFamily": "Roboto",
          "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
          "spacing": Object {
            "lg": "4em",
            "md": "2em",
            "nm": "1em",
            "none": "0em",
            "sm": "0.5em",
            "xl": "8em",
          },
        }
      }
    >
      <ForwardRef
        bg="blue"
        border={0}
        borderRadius={2}
        className="test c0 c1"
        color="white"
        fontSize={1}
        fontWeight="bold"
        is="button"
        lineHeight={1.1428571428571428}
        m={0}
        px={3}
        py={2}
        theme={
          Object {
            "colors": Object {
              "addOnTab": "#347CFF",
              "addToCart": "#347CFF",
              "addonBg": "#EEEEEE",
              "backgroundGrey": "#E6E6E6",
              "black": "#000000",
              "close": "red",
              "contextBackground": "#f0f0f3",
              "darkGrey": "#464646",
              "disableBackground": "#E4E4E4",
              "disableButton": "#D1D1D1",
              "error": "#cc0000",
              "footer": "#48a942",
              "glutenFree": "#8d7356",
              "healthy": "#359919",
              "hint": "#a6a6a6",
              "inputBottomBorder": "lightgray",
              "itemBorder": "lightgrey",
              "itemTile": "#3C474C0",
              "light": "white",
              "loader": "#ababab",
              "loginBackground": "#303030",
              "loginBoxColor": "#434149",
              "loginField": "#AAA7AC",
              "loginFormBackground": "#484848",
              "navigationMenu": "#48a942",
              "navigationMenuHover": "cornflowerblue",
              "new": "#ff6000",
              "open": "#48a942",
              "paymentSuccessTitle": "#505050",
              "placeHolder": "#909090",
              "primary": "#347CFF",
              "primaryButton": "#347CFF",
              "primaryText": "#383838",
              "secondaryText": "#a6a6a6",
              "spicy": "#ed1c24",
              "text": "black",
              "textGrey": "#505050",
              "tile": "#505050",
              "validationError": "#cc0000",
              "vegan": "#0d4a22",
              "vegetarian": "#0d4a22",
            },
            "fontFamily": "Roboto",
            "fontSize": Object {
              "lg": "1.8rem",
              "md": "1.3rem",
              "nm": "1rem",
              "sm": "0.75rem",
              "xl": "2.4rem",
            },
            "fontSizeFromProps": [Function],
            "mediaBreakpoints": Object {
              "desktop": [Function],
              "mobile": [Function],
              "tablet": [Function],
            },
            "nav": Object {
              "height": "70px",
            },
            "receiptFontFamily": "Roboto",
            "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
            "spacing": Object {
              "lg": "4em",
              "md": "2em",
              "nm": "1em",
              "none": "0em",
              "sm": "0.5em",
              "xl": "8em",
            },
          }
        }
      >
        <button
          className="test c0 c1"
          theme={
            Object {
              "colors": Object {
                "addOnTab": "#347CFF",
                "addToCart": "#347CFF",
                "addonBg": "#EEEEEE",
                "backgroundGrey": "#E6E6E6",
                "black": "#000000",
                "close": "red",
                "contextBackground": "#f0f0f3",
                "darkGrey": "#464646",
                "disableBackground": "#E4E4E4",
                "disableButton": "#D1D1D1",
                "error": "#cc0000",
                "footer": "#48a942",
                "glutenFree": "#8d7356",
                "healthy": "#359919",
                "hint": "#a6a6a6",
                "inputBottomBorder": "lightgray",
                "itemBorder": "lightgrey",
                "itemTile": "#3C474C0",
                "light": "white",
                "loader": "#ababab",
                "loginBackground": "#303030",
                "loginBoxColor": "#434149",
                "loginField": "#AAA7AC",
                "loginFormBackground": "#484848",
                "navigationMenu": "#48a942",
                "navigationMenuHover": "cornflowerblue",
                "new": "#ff6000",
                "open": "#48a942",
                "paymentSuccessTitle": "#505050",
                "placeHolder": "#909090",
                "primary": "#347CFF",
                "primaryButton": "#347CFF",
                "primaryText": "#383838",
                "secondaryText": "#a6a6a6",
                "spicy": "#ed1c24",
                "text": "black",
                "textGrey": "#505050",
                "tile": "#505050",
                "validationError": "#cc0000",
                "vegan": "#0d4a22",
                "vegetarian": "#0d4a22",
              },
              "fontFamily": "Roboto",
              "fontSize": Object {
                "lg": "1.8rem",
                "md": "1.3rem",
                "nm": "1rem",
                "sm": "0.75rem",
                "xl": "2.4rem",
              },
              "fontSizeFromProps": [Function],
              "mediaBreakpoints": Object {
                "desktop": [Function],
                "mobile": [Function],
                "tablet": [Function],
              },
              "nav": Object {
                "height": "70px",
              },
              "receiptFontFamily": "Roboto",
              "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
              "spacing": Object {
                "lg": "4em",
                "md": "2em",
                "nm": "1em",
                "none": "0em",
                "sm": "0.5em",
                "xl": "8em",
              },
            }
          }
        />
      </ForwardRef>
    </Button>
  </Styled(Button)>
</PrimaryButton>
`;
